nomos-services/data-availability/indexer/src/lib.rs

```diff
@@ -0,0 +1,474 @@
+    DaPoolAdapter,
+    DaVerificationProvider,
+    TxS,
+    BS,
+    ConsensusStorage,
+> where
+    Indexer: DaIndexer,
+    Indexer::Blob: 'static,
+    Indexer::VID: 'static,
+    A: NetworkAdapter,
+    ClPoolAdapter: MempoolAdapter<Payload = ClPool::Item, Key = ClPool::Key>,
+    ClPool: MemPool<BlockId = HeaderId>,
+    DaPool: MemPool<BlockId = HeaderId>,
+    DaPoolAdapter: MempoolAdapter<Key = DaPool::Key>,
+    DaPoolAdapter::Payload: Certificate + Into<DaPool::Item> + Debug,
+    DaVerificationProvider: MempoolVerificationProvider<
+        Payload = DaPoolAdapter::Payload,
+        Parameters = <DaPoolAdapter::Payload as Certificate>::VerificationParameters,
+    >,
+    ClPool::Item: Clone + Eq + Hash + Debug + 'static,
+    ClPool::Key: Debug + 'static,
+    DaPool::Item: Clone + Eq + Hash + Debug + 'static,
+    DaPool::Key: Debug + 'static,
+    A::Backend: 'static,
+    TxS: TxSelect<Tx = ClPool::Item>,
+    BS: BlobCertificateSelect<Certificate = DaPool::Item>,
+    DaStorage: DaStorageAdapter,
+    ConsensusStorage: StorageBackend + Send + Sync + 'static,
+{
+    service_state: ServiceStateHandle<Self>,
+    indexer: Indexer,
+    storage_relay: Relay<StorageService<DaStorage::Backend>>,
+    consensus_relay: ConsensusRelay<
+        A,
+        ClPool,
+        ClPoolAdapter,
+        DaPool,
+        DaPoolAdapter,
+        DaVerificationProvider,
+        TxS,
+        BS,
+        ConsensusStorage,
+    >,
+}
+
+impl<
+        Indexer,
+        DaStorage,
+        Consensus,
+        A,
+        ClPool,
+        ClPoolAdapter,
+        DaPool,
+        DaPoolAdapter,
+        DaVerificationProvider,
+        TxS,
+        BS,
+        ConsensusStorage,
+    >
+    DataIndexerService<
+        Indexer,
+        DaStorage,
+        Consensus,
+        A,
+        ClPool,
+        ClPoolAdapter,
+        DaPool,
+        DaPoolAdapter,
+        DaVerificationProvider,
+        TxS,
+        BS,
+        ConsensusStorage,
+    >
+where
+    Indexer: DaIndexer,
+    Indexer::Blob: 'static,
+    Indexer::VID: 'static,
+    A: NetworkAdapter,
+    ClPoolAdapter: MempoolAdapter<Payload = ClPool::Item, Key = ClPool::Key>,
+    ClPool: MemPool<BlockId = HeaderId>,
+    DaPool: MemPool<BlockId = HeaderId>,
+    DaPoolAdapter: MempoolAdapter<Key = DaPool::Key>,
+    DaPoolAdapter::Payload: Certificate + Into<DaPool::Item> + Debug,
+    DaVerificationProvider: MempoolVerificationProvider<
+        Payload = DaPoolAdapter::Payload,
+        Parameters = <DaPoolAdapter::Payload as Certificate>::VerificationParameters,
+    >,
+    ClPool::Item: Clone + Eq + Hash + Debug + 'static,
+    ClPool::Key: Debug + 'static,
+    DaPool::Item: Clone + Eq + Hash + Debug + 'static,
+    DaPool::Key: Debug + 'static,
+    A::Backend: 'static,
+    TxS: TxSelect<Tx = ClPool::Item>,
+    BS: BlobCertificateSelect<Certificate = DaPool::Item>,
+    DaStorage: DaStorageAdapter,
+    ConsensusStorage: StorageBackend + Send + Sync + 'static,
+{
+}
```

Comment:
This impl block is empty 🤔

Reply:
Hm, not sure how compiler didn't catch this, because there's another `DaIndexerService` implementation with `handle_new_block` and `handle_da_msg` methods.

Reply:
In rust you can have many impl blocks for the same item.

Reply:
50 LOCs of empty implementation removed.

Reply:
🍬 sweet 🍬